File: Topalian_Python_Text_File_Journal/pdf/combine/Topalian_Combine_txt_Files.py
# Topalian_Combine_txt_Files.py

import os

def combineTextFiles(directory, textFileName):
    outputFilePath = os.path.join(directory, 'main.txt')
    fileContents = []

    def traverseFolder(folder):
        for root, dirs, files in os.walk(folder):
            for file in files:
                filePath = os.path.join(root, file)
                if filePath != textFileName and filePath.endswith('.txt'):
                    with open(filePath, 'r', encoding='utf-8') as f:
                        fileContents.append(f.read())

    traverseFolder(directory)

    with open(outputFilePath, 'w', encoding='utf-8') as f:
        f.write('\n'.join(fileContents))
    print(f"Combined {len(fileContents)} .txt files into {outputFilePath}")

# get current directory of script
currentDirectory = os.path.dirname(os.path.abspath(__file__))

# get filename of script
textFileName = os.path.abspath(__file__)

combineTextFiles(currentDirectory, textFileName)

File: Topalian_Python_Text_File_Journal/pdf/combine/Topalian_Combine_txt_Files.js
// Topalian_Combine_txt_Files.js

let fs = require('fs');
let path = require('path');

function combineTextFiles(directory, textFileName)
{
    let outputFilePath = path.join(directory, 'main.txt');

    let fileContents = [];

    function traverseFolder(folder)
    {
        let files = fs.readdirSync(folder);

        for (let i = 0; i < files.length; i++)
        {
            let file = files[i];

            let filePath = path.join(folder, file);

            let stats = fs.statSync(filePath);

            if (stats.isDirectory())
            {
                traverseFolder(filePath);
            }
            else if (path.extname(filePath) === '.txt')
            {
                let content = fs.readFileSync(filePath, 'utf8');
                // check if file is not script file itself
                if (filePath !== textFileName)
                {
                    fileContents.push(content);
                }
            }
        }
    }

    traverseFolder(directory);

    fs.writeFileSync(outputFilePath, fileContents.join('\n'), 'utf8');

    console.log(`Combined ${fileContents.length} .txt files into ${outputFilePath}`);
}

// get current directory of script
let currentDirectory = process.cwd();

// get filename of script
let textFileName = __filename;

combineTextFiles(currentDirectory, textFileName);